devices/usbc-cdc/ctrl.c

```diff
@@ -65,49 +71,49 @@ static int ctrl_allocBuff(int endpt, int dir)
 
 static void *ctrl_allocQtdMem(void)
 {
-	int i;
-
-	if (!ctrl_common.qtdOffs) {
-		ctrl_common.dc->dtdMem = usbclient_allocBuff(USB_BUFFER_SIZE);
-
-		if (ctrl_common.dc->dtdMem == NULL)
-			return NULL;
+	if (ctrl_common.qtdOffs == 0) {
+		/* Allocate buffer for the first time (initially dtdMem is NULL) */
+		if (ctrl_common.dc->dtdMem == NULL) {
+			ctrl_common.dc->dtdMem = usbclient_allocBuff(USB_BUFFER_SIZE);
+			if (ctrl_common.dc->dtdMem == NULL) {
+				return NULL;
+			}
+		}
 
-		for (i = 0; i < USB_BUFFER_SIZE; ++i)
-			((char *)ctrl_common.dc->dtdMem)[i] = 0;
+		hal_memset(ctrl_common.dc->dtdMem, 0, USB_BUFFER_SIZE);
 	}
 
-	return (void *)(ctrl_common.dc->dtdMem + 0x40 * (ctrl_common.qtdOffs++));
+	return (void *)(ctrl_common.dc->dtdMem + (u32)64 * (ctrl_common.qtdOffs++));
```

Comment:
```suggestion
	return (void *)(ctrl_common.dc->dtdMem + 64U * (ctrl_common.qtdOffs++));
```

Reply:
Anyway, why was this change needed? I see some 64s (previously 0x40s) in other places that are not unsigned

Reply:
Much of this code tends to use the magic values 64, 0x40, etc (of course they both are equal). For me, using hexadecimal values is ok where it's ok, in the sense of using them consistently and not messing around. The 64 you are asking about is from the documentation, there is always a decimal value when it comes to the number of bytes, and it would look strange in human documentation to write "0x4d2 bytes" instead of 1234 bytes.
And finally comes the code search for "magic values", you expect 1234 as in the reference manual and someone wrote down 0x4d2 (a magic value), just a matter of taste.

Cast to `u32` (but better suits `uintptr_t`) it comes from that the result of multiplication is in type 'int' and is used as a pointer offset after an implicit widening conversion to type 'ptrdiff_t', here `qtdOffs` is unsigned and `+` binary operator was used, I expect that in effect address is unsigned (addition), but while 64 is an integer it is promoted.  Depending on (void*) size proper cast should be used, here looks like neither `u32` is ok nor `U` suffix because of portability, but cast to `uintptr_t`, but plo does not define it so maybe `addr_t` is better at the moment ?